bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableVeneerSettingsFactory.java

```diff
@@ -0,0 +1,327 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.config;
+
+import static com.google.api.client.util.Preconditions.checkState;
+import static com.google.cloud.bigtable.config.CallOptionsConfig.SHORT_TIMEOUT_MS_DEFAULT;
+import static io.grpc.internal.GrpcUtil.USER_AGENT_KEY;
+import static org.threeten.bp.Duration.ofMillis;
+
+import com.google.api.core.ApiFunction;
+import com.google.api.gax.batching.BatchingSettings;
+import com.google.api.gax.batching.FlowControlSettings;
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.api.gax.core.NoCredentialsProvider;
+import com.google.api.gax.grpc.GrpcStatusCode;
+import com.google.api.gax.grpc.InstantiatingGrpcChannelProvider;
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.api.gax.rpc.FixedHeaderProvider;
+import com.google.api.gax.rpc.HeaderProvider;
+import com.google.api.gax.rpc.StatusCode;
+import com.google.api.gax.rpc.TransportChannelProvider;
+import com.google.auth.Credentials;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminSettings;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings.Builder;
+import com.google.common.collect.ImmutableSet;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Status;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Set;
+import javax.annotation.Nonnull;
+import org.threeten.bp.Duration;
+
+/**
+ * Static methods to convert an instance of {@link BigtableOptions} to a
+ * {@link BigtableDataSettings} or {@link BigtableTableAdminSettings} instance .
+ */
+public class BigtableVeneerSettingsFactory {
+
+  /** Constant <code>LOG</code> */
+  private static final Logger LOG = new Logger(BigtableVeneerSettingsFactory.class);
+
+  //Identifier to distinguish between CBT or GCJ adapter.
+  private static final String VENEER_ADAPTER =  BigtableVersionInfo.CORE_USER_AGENT+","
+      + "VENEER_ADAPTER";
+
+  // 256 MB, server has 256 MB limit.
+  private final static int MAX_MESSAGE_SIZE = 1 << 28;
+
+  /**
+   * To create an instance of {@link BigtableDataSettings} from {@link BigtableOptions}.
+   *
+   * @param options a {@link BigtableOptions} object.
+   * @return a {@link BigtableDataSettings} object.
+   * @throws IOException if any.
+   */
+  public static BigtableDataSettings createBigtableDataSettings(@Nonnull final BigtableOptions options) throws IOException {
+    checkState(options.getProjectId() != null, "Project ID is required");
+    checkState(options.getInstanceId() != null, "Instance ID is required");
+    checkState(options.getRetryOptions().enableRetries(), "Disabling retries is not currently supported.");
```

Comment:
I don't think useCachedDataPool option is supported here, please throw an error if its enabled

Reply:
Added state check for useCachedChannel:
```java
checkState(!options.useCachedChannel(), "cachedDataPool is not currently supported.");
````